feat: Subscriptions with Stripe function (#2)
appwrite/templates: node/subscriptions-with-stripe/README.md

# 💳 Node.js Stripe Subscriptions Function

> Integrates Stripe subscriptions into your Appwrite project. Collect card payment with the `/checkout` endpoint and check the status of a user subscription using the `Subscriptions` collection.

Reviewer: ```suggestion
Integrates Stripe subscriptions into your Appwrite project. Collect card payment with the `/subscribe` endpoint and check the status of a user subscription using the `Subscriptions` collection.
```

| failureUrl         | The URL to redirect to after a cancelled payment attempt. | Body   | String | https://example.com/failure    |

**Response**

Sample `303` Response:

The response is a redirect to the Stripe checkout session URL or to the cancel URL if an error occurs

```text
Location: https://checkout.stripe.com/pay/cs_test_...#fidkdWxOYHwnP
```

```text
Location: https://mywebapp.com/cancel
```

### `POST /webhook`

This endpoint is a webhook that handles two types of events from Stripe: `customer.subscription.created` and `customer.subscription.deleted`. It validates the incoming request using the Stripe's validateWebhook method. If the validation fails, a `401` response is sent.

**Parameters**

| Name | Description                  | Location | Type   | Sample Value                                                          |
| ---- | ---------------------------- | -------- | ------ | --------------------------------------------------------------------- |
| None | Webhook payload from Stripe. | Body     | Object | [See Stripe documentation](https://stripe.com/docs/api/events/object) |

**Response**

Sample `200` Response:

> In case of `customer.subscription.created` event, it creates a new subscription for the user.
> In case of `customer.subscription.deleted` event, it deletes the subscription for the user.

Reviewer: ```suggestion
In case of `customer.subscription.created` event, it gives user `subscriber` label.
In case of `customer.subscription.deleted` event, it takes `subscriber` label away from user.
```

```json
{ "success": true }
```

Sample `401` Response:

```json
{ "success": false }
```

## ⚙️ Configuration

| Setting           | Value           |
| ----------------- | --------------- |
| Runtime           | Node (18.0)     |
| Entrypoint        | `src/main.js`   |
| Build Commands    | `npm install`   |
| Permissions       | `any`           |
| Timeout (Seconds) | 15              |

> 

Reviewer: ```suggestion

> If using a demo web app to subscribe, make sure to add your function domain as a web platform to your Appwrite project. Doing this fixes CORS errors and allows proper functionality.
```

# 💳 Node.js Stripe Subscriptions Function

> Integrates Stripe subscriptions into your Appwrite project. Collect card payment with the `/subscribe` endpoint and check the status of a user subscription using the `Subscriptions` collection.

Reviewer: ```suggestion
Receive recurring payments and grant subscribers extra permissions.
```

> Integrates Stripe subscriptions into your Appwrite project. Collect card payment with the `/subscribe` endpoint and check the status of a user subscription using the `Subscriptions` collection.

Reviewer: ```suggestion
Receive recurring card payments and grant subscribers extra permissions.
```
